src/Path/Extra.hs

```diff
@@ -43,6 +44,12 @@ parseCollapsedAbsDir = parseAbsDir . collapseFilePath
 parseCollapsedAbsFile :: MonadThrow m => FilePath -> m (Path Abs File)
 parseCollapsedAbsFile = parseAbsFile . collapseFilePath
 
+-- | Combine a 'Path Abs Dir' and a relative 'FilePath' into a 'Path Abs Dir'
```

Comment:
Heh. It makes me a bit uncomfortable as well. Perhaps it is better to be more colloquial and terse?
`+-- | Add a relative FilePath to the end of a Path`